Remove deprecated uppercase URL attribute alias from EventSource interface

Ahmad-S792 opened the pull request:
#### 710c43c6b66e8eeb73f6eed984155b43c2936390
<pre>
Remove deprecated uppercase URL attribute alias from EventSource interface

Remove deprecated uppercase URL attribute alias from EventSource interface
<a href="https://bugs.webkit.org/show_bug.cgi?id=248787">https://bugs.webkit.org/show_bug.cgi?id=248787</a>

Reviewed by Ryosuke Niwa.

This is to align Webkit with Blink / Chromium, Gecko / Firefox and Web-Specifications:

<a href="https://html.spec.whatwg.org/multipage/server-sent-events.html#the-eventsource-interface">https://html.spec.whatwg.org/multipage/server-sent-events.html#the-eventsource-interface</a>

Partial Merge - <a href="https://chromium.googlesource.com/chromium/blink/+/f0e6045bfaa96d3a3211e8074feb83d37913f9d3">https://chromium.googlesource.com/chromium/blink/+/f0e6045bfaa96d3a3211e8074feb83d37913f9d3</a>

The upper case &quot;URL&quot; attribute was only in draft specifications from 2009 but later dropped in 2011 specs, which were implemented by non-Webkit browsers. It was later dropped by Blink in 2015 as well.

* Source/WebCore/page/EventSource.idl: Remove &apos;URL&apos; attribute and also add link to web-spec
* LayoutTests/http/tests/eventsource/eventsource-url-attribute.html: Rebaselined
* LayoutTests/http/tests/eventsource/eventsource-url-attribute-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/257553@main">https://commits.webkit.org/257553@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/82f0b9f1aa2ac1319444f62e4f8a7dcbf24d66bc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99146 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8352 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32284 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108541 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168783 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103144 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8915 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85692 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91652 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106469 "Hash 82f0b9f1 for PR 7194 does not build (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104883 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6747 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90309 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33746 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88571 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21655 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76610 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2241 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23172 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2133 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45579 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5194 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/7135 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42647 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3566 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on LayoutTests/http/tests/eventsource/eventsource-url-attribute.html:
```diff
@@ -7,14 +7,12 @@
 
-shouldBeEqualToString("source.URL", url);
 shouldBeEqualToString("source.url", url);
-shouldBeTrue("source.URL === source.url");
```
We should probably assert `shouldBeUndefined("source.URL")`

Ahmad-S792 replied:
Done.

rniwa reviewed (approved):
r=me

Ahmad-S792 committed:
Remove deprecated uppercase URL attribute alias from EventSource interface

Remove deprecated uppercase URL attribute alias from EventSource interface
https://bugs.webkit.org/show_bug.cgi?id=248787

Reviewed by Ryosuke Niwa.

This is to align Webkit with Blink / Chromium, Gecko / Firefox and Web-Specifications:

https://html.spec.whatwg.org/multipage/server-sent-events.html#the-eventsource-interface

Partial Merge - https://chromium.googlesource.com/chromium/blink/+/f0e6045bfaa96d3a3211e8074feb83d37913f9d3

The upper case "URL" attribute was only in draft specifications from 2009 but later dropped in 2011 specs, which were implemented by non-Webkit browsers. It was later dropped by Blink in 2015 as well.

* Source/WebCore/page/EventSource.idl: Remove 'URL' attribute and also add link to web-spec
* LayoutTests/http/tests/eventsource/eventsource-url-attribute.html: Rebaselined
* LayoutTests/http/tests/eventsource/eventsource-url-attribute-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/257553@main
